### DISCUSSION TITLE
Diverged line searches after a restart

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#20167] [Q&A General] [answered]
I have a fairly large input file with a lot of moving parts that converges just fine and gives the correct output.

I then take the output as an initial condition for precisely the same input file and I get diverged line searches on the first time step after the second nonlinear iteration. 

I find that if I increase my time step to around where it was when the original calculation was used, it converges no problem. But if it is too small, then I get diverged line searches.

Has anyone encountered this before?

**Comment by GiudGiud** [comment, marked as answer]
Hello

does it look like it s converging between 1st and 2nd iteration?
Does it converge without a line search? ‘Line_search=none’ in the Executioner parameter
Did you restart everything, even the auxvaruables?
Are you using the default 1st order implicit time scheme?

Guillaume

**Reply by mangerij** [reply]
Yes, converges perfectly between first and second nonlinear iterations.

Turning line search off does not let it converge.

I am restarting all auxvariables yes.

And yes, using implicit-euler.

And I should mention that I avoided this problem by taking larger time steps.  But I found this issue to be strange to me and took me a while to figure it out.

**Reply by GiudGiud** [reply]
@fdkong does the default line search have some sort of history / informed by previous searches feature?

**Reply by fdkong** [reply]
> @fdkong does the default line search have some sort of history / informed by previous searches feature?

No, linesearch does not store history. 

The issue happened here might be caused by an incomplete restart. If we restart correctly, the nonlinear and linear residuals should be the same.

We could enable checkpoint files and restart from the previous time step for debugging. Then compare the solver behaviors of the restarted simulation with the original one at the same time step.

>I then take the output as an initial condition for precisely the same input file

@mangerij 
What kind of files you restart with? Exodus or check point?

**Reply by mangerij** [reply]
Exodus. What is a checkpoint file? Sorry I cannot keep up to date with all the goodies in the framework.

By the way, this is probably what happened? Since I am testing something and didn't set an end to the simulation, I just killed the solve by ctrl+C to terminal.

**Reply by GiudGiud** [reply]
See this page
https://mooseframework.inl.gov/application_usage/restart_recover.html
Checkpoint is the more official way of restarting simulations. 

So the exodus was generated after a Ctrl+C ? You could have been unlucky there and stopped it during writing

**Reply by fdkong** [reply]
> Exodus.

Then I won't be surprised you got different behaviors by restarting from Exodus.  You only got the solution back from Exodus, but you lost other things.

A better way to do restart is to use checkpoint.

**Reply by fdkong** [reply]
>So the exodus was generated after a Ctrl+C ? You could have been unlucky there and stopped it during writing

Not only that. checkpoint has extra information to restart the exact simulation ...
